update.ps1

```diff
@@ -1,201 +1,532 @@
+
+        if ($actionContext.Configuration.updateManagerOnUpdate -eq $true) {
+            #region Get manager of account
+            # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: GET /api/users/{identifier}/manager
+            $actionMessage = "querying manager of account with AccountReference: $($actionContext.References.Account | ConvertTo-Json)"
+    
+            $getKPNLisaAccountManagerSplatParams = @{
+                Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($actionContext.References.Account)/manager"
+                Method      = "GET"
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
+    
+            Write-Verbose "SplatParams: $($getKPNLisaAccountSplatParams | ConvertTo-Json)"
+    
+            # Add header after printing splat
+            $getKPNLisaAccountManagerSplatParams['Headers'] = $headers
+    
+            $getKPNLisaAccountManagerResponse = $null
+            $getKPNLisaAccountManagerResponse = Invoke-RestMethod @getKPNLisaAccountManagerSplatParams
+            $previousManagerId = $getKPNLisaAccountManagerResponse.id
+            
+            Write-Verbose "Queried manager of account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Result: $($getKPNLisaAccountManagerResponse | ConvertTo-Json)"
+            #endregion Get manager of  account
+    
+            #region Calulate manager action
+            if ($actionContext.References.ManagerAccount -ne $previousManagerId) {
+                if (-not[String]::IsNullOrEmpty(($actionContext.References.ManagerAccount))) {
+                    $actionManager = "Update"
+                }
+                else {
+                    $actionManager = "Clear"
+                }
+            }
+            else {
+                $actionManager = "NoChanges"
+            }
+            #endregion Calulate manager action
 
-    Write-Verbose -Verbose "Updating KPN Lisa account for '$($personContext.Person.DisplayName)'"
+            # Add managerId to Previous data
+            $outputContext.PreviousData | Add-Member -MemberType NoteProperty -Name "managerId" -Value $previousManagerId -Force
+        }
 
-    $SplatParams = @{
-        Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($actionContext.References.Account)/bulk"
-        Method = "Patch"
-        Body   = $outputContext.Data | Select-Object -Property * -ExcludeProperty $NonUpdatables
-    }
+        # Create reference object from correlated account
+        $accountReferenceObject = [PSCustomObject]@{}
+        foreach ($correlatedAccountProperty  in ($correlatedAccount | Get-Member -MemberType NoteProperty)) {
+            # Add property using -join to support array values
+            $accountReferenceObject | Add-Member -MemberType NoteProperty -Name $correlatedAccountProperty.Name -Value ($correlatedAccount.$($correlatedAccountProperty.Name) -join ",") -Force
```

Comment:
Hier mogelijk nog rekening houden met $null values. Anders kan de -join misgaan.

Reply:
Goed punt, maar gelukkig veroorzaakt een $null value geen error bij de join. Dan is je resultaat gewoon $null. 
Dit is ook getest met $null waardes op het account en is dan ook geen issue.